include/Value.h

```diff
@@ -82,23 +82,18 @@ class value_type_large {
 public:
 	template <class U>
 	static value_type_large fromZeroExtended(const U &data) {
-		value_type_large created;
-		created.copyZeroExtended(data);
-		return created;
-	}
 
-private:
-	template <class U>
-	void copyZeroExtended(const U& data) {
-		// TODO(eteran): this seems to be more complex than I think we need
-		// since this is an unsigned type, I suspect that just assigning would be enough so long as it would fit...
 		static_assert(sizeof(U) <= sizeof(T), "It doesn't make sense to expand a larger type into a smaller type");
 
+		value_type_large created;
+
 		auto dataStart = reinterpret_cast<const char*>(&data);
-		auto target    = reinterpret_cast<char*>(&value_);
+		auto target    = reinterpret_cast<char*>(&created.value_);
+
+		std::memcpy(target, dataStart, sizeof(U));
+		std::memset(target + sizeof(U), 0, sizeof(T) - sizeof(U));
```

Comment:
It's better to refer to objects when doing offset calculations than to some (possibly unrelated) types. It'll 1) add more clarity (show why we're using this size and not another), and 2) will automatically update the sizeofs to refer to the same objects even if their types change.

Reply:
I looked through your comments, and they generally all make sense :-). I'll implement some of the changes you suggest!